lib/Provider/MailProvider.php

```diff
@@ -0,0 +1,238 @@
+<?php
+
+declare(strict_types=1);
+
+/**
+ * SPDX-FileCopyrightText: 2024 Nextcloud GmbH and Nextcloud contributors
+ * SPDX-License-Identifier: AGPL-3.0-or-later
+ */
+namespace OCA\Mail\Provider;
+
+use OCA\Mail\Account;
+use OCA\Mail\Service\AccountService;
+
+use OCP\Mail\Provider\Address as MailAddress;
+use OCP\Mail\Provider\IProvider;
+use OCP\Mail\Provider\IService;
+
+use Psr\Container\ContainerInterface;
+
+class MailProvider implements IProvider {
+
+	private ?array $ServiceCollection = [];
+
+	public function __construct(
+		protected ContainerInterface $container,
+		protected AccountService $AccountService
+	) {
+	}
+
+	/**
+	 * arbitrary unique text string identifying this provider
+	 *
+	 * @since 2024.05.25
```

Comment:
```suggestion
	 * @since 4.0.0
```

Please use the mail version this change will be released with (if any)